desc/vmec_utils.py

```diff
@@ -469,3 +487,509 @@ def vmec_boundary_subspace(eq, RBC=None, ZBS=None, RBS=None, ZBC=None):  # noqa:
     boundary_subspace = block_diag(Rb_subspace, Zb_subspace)
     opt_subspace = null_space(boundary_subspace)
     return opt_subspace
+
+
+def make_boozmn_output(  # noqa: 16 fxn too complex
+    eq, path, surfs=128, M_booz=None, N_booz=None, verbose=0
+):
+    """Create and save a booz_xform-style .nc output file.
+
+    based strongly off of https://github.com/hiddenSymmetries/booz_xform/tree/main
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium to save.
+    path : str
+        File path of output data.
+    surfs: int
+        Number of flux surfaces to calculate Boozer transform at (Default = 128).
+        NOTE: because this is performed on the so-called "half-grid", this will
+        result in an output of size surfs-1, since for surfs number of surfaces
+        there is only surfs-1 surfaces on the half-grid, the first one being
+        the surface at s = 0.5 / surfs
+        where s = rho**2 is the normalized toroidal flux coordinate
+    M_booz : int, optional
+        poloidal resolution to use for Boozer transform.
+    N_booz : int, optional
+        toroidal resolution to use for Boozer transform.
+    verbose: int
+        Level of output (Default = 1).
+        * 0: no output
+        * 1: status of quantities computed
+        * 2: as above plus timing information
+
+    Returns
+    -------
+    None
+
+    """
+    timer = Timer()
+    timer.start("Total time")
+
+    Psi = eq.Psi
+    NFP = eq.NFP
+    if M_booz is None:
+        M_booz = 2 * eq.M
+    if N_booz is None:
+        N_booz = 2 * eq.N
+
+    # calculations are done on a "half-grid" of size surfs-1
+    # which starts at s =  psi = 0.5/(surfs-1)
+    # and increments by 1 / (surfs-1)
+    # until it reaches 1-0.5/(surfs-1)
+    # VMEC radial coordinate: s = rho^2 = Psi / Psi(LCFS)
+    s_full = np.linspace(0, 1, surfs)
+    hs = 1 / (surfs - 1)
+    s_half = np.arange(hs / 2, 1, hs)
+    r_full = np.sqrt(s_full)
+    r_half = np.sqrt(s_half)
+
+    basis = get_transforms(
```

Comment:
would it be easier to just create the double fourier basis here, since thats all you need? Otherwise it creates a bunch of other stuff that gets thrown out

Reply:
ended up changing so that I use the transforms as well